tests/cli/test_units.py

```diff
@@ -141,3 +143,43 @@ def test_netplan_try_is_not_revertable(self):
 ''')
         cmd = NetplanTry()
         self.assertFalse(cmd.is_revertable())
+
+    def test_raises_exception_main_function(self):
+        with open(os.path.join(self.tmproot, 'etc/netplan/test.yaml'), 'w') as f:
+            f.write('''network:
+              ethernets:
+                eth0:
+                  dhcp4: nothanks''')
+
+        # The idea was to capture stderr here but for some reason
+        # my attempts to mock sys.stderr didn't work with pytest
```

Comment:
note: for the record, many test harnesses mess with std{err,out} to be able to capture output for failing tests while suppressing it for the rest, in order to keep a clean run output. It's not surprising that it conflicts with standard mocking techniques.

There seem to be a way to access the captured output with pytest though: https://docs.pytest.org/en/7.1.x/how-to/capture-stdout-stderr.html#accessing-captured-output-from-a-test-function

I'm happy with the alternative you came up with, though :)